src/app/page.tsx

```diff
@@ -117,6 +123,26 @@ const Home: React.FC<HomeProps> = () => {
     },
   });
 
+  const getAllFinalityProviders = useCallback(async () => {
```

Comment:
Not a big fan, but it looks like it's the only option we can use? With the UI request, given the API we have and BD requirement?

What I mean: we have a react-query `data: finalityProviders`. Then we have `getAllFinalityProviders` as callback, that does `setState` in `useEffect`

As "good enough" approach: it's okay. But is there a way to simplify this fetching logic? Looks like we're in somewhat "dirty" zone when we fetch the FPs using `react-query`, `useCallback`, `useEffect`, `setState` and then `useMemo`.

Looking at the code for me, it was somehow strange to wrap my head around of what's going on there.

If "good enough" approach is "enough" right now, or we can't go with the "simpler" approach: I am fine with that logic. Although indeed this will be extracted at some point (when we refactor `page.tsx`) and I am almost sure that this will (at least it **should**) be simplified.